llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -366,6 +379,10 @@ template <typename... Preds> Or<Preds...> m_AnyOf(Preds &&...preds) {
   return Or<Preds...>(std::forward<Preds>(preds)...);
 }
 
+template <typename... Preds> auto m_NoneOf(Preds &&...preds) {
+  return Not{m_AnyOf(std::forward<Preds>(preds)...)};
```

Comment:
Maybe we want to have something like `m_not()` to wrap the struct instantiation?

Reply:
Done. I created a `m_IsNot` to wrap the `Not` struct instantiation.